src/diffusers/schedulers/scheduling_dpmsolver_multistep.py

```diff
@@ -880,6 +930,11 @@ def step(
         if self.step_index is None:
             self._init_step_index(timestep)
 
+        # store old dtype because model_output isn't always the same it seems
+        return_dtype = sample.dtype
```

Comment:
do you mean `model_output.dtype` isn't always the same as `sample.dtype` before the upcast?

Reply:
Turns out that was because `model_output = self.convert_model_output(model_output, sample=sample)` ends up casting it to the sample's dtype. I moved the sample upcast after this call so `return_type` is no longer needed. Outputs are the same.